optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -679,8 +683,8 @@ private void determineGlobalShadowOrder() {
     /**
      * @return the {@link Class} of {@link PlanningScore}
      */
-    public Class<? extends Score> extractScoreClass() {
-        return (Class<? extends Score>) scoreMemberAccessor.getType();
+    public Class<? extends Score<?>> extractScoreClass() {
```

Comment:
Now wouldn't it be nice if `SolutionDescriptor` also had a `Score_`? But maybe that'd just go too far.

Reply:
I fear this will go to far. That's a rabbit hole that will lead very deep.
Separate PR/experiment in any case. And let's first figure out if InnerScoreDirector should be score generified...